packages/lit-element/src/lit-element.ts

```diff
@@ -127,127 +105,16 @@ export class LitElement extends UpdatingElement {
-  readonly _renderOptions!: RenderOptions;
+  readonly _renderOptions: RenderOptions = {eventContext: this};
 
-  /**
-   * Performs element initialization. By default this calls
-   * [[`createRenderRoot`]] to create the element [[`renderRoot`]] node and
-   * captures any pre-set values for registered properties.
-   */
-  protected initialize() {
-    super.initialize();
-    (this as {
-      _renderOptions: RenderOptions;
-    })._renderOptions = {eventContext: this};
-    (this as {
-      renderRoot: Element | DocumentFragment;
-    }).renderRoot = this.createRenderRoot();
-    this.adoptStyles((this.constructor as typeof LitElement)._elementStyles!);
-  }
-
-  /**
-   * Returns the node into which the element should render and by default
-   * creates and returns an open shadowRoot. Implement to customize where the
-   * element's DOM is rendered. For example, to render into the element's
-   * childNodes, return `this`.
-   * @returns {Element|DocumentFragment} Returns a node into which to render.
-   */
-  protected createRenderRoot(): Element | ShadowRoot {
-    return this.attachShadow(
-      (this.constructor as typeof LitElement).shadowRootOptions
-    );
-  }
-
-  /**
-   * Applies the given styles to the element. Styling is applied to the element
-   * only if the `renderRoot` is a `shadowRoot`. If the `rendeRoot` is not
-   * a `shadowRoot`, this method may be overridden to apply styling in another
-   * way. Styling will apply using `shadowRoot.adoptedStyleSheets` where
-   * available and will fallback otherwise. When Shadow DOM is available but
-   * `adoptedStyleSheets` is not, styles are appended to the the `shadowRoot`
-   * to [mimic spec behavior](https://wicg.github.io/construct-stylesheets/#using-constructed-stylesheets).
-   */
-  protected adoptStyles(styles: CSSResultFlatArray) {
-    // Note, if renderRoot is not a shadowRoot, styles would/could apply to the
-    // element's getRootNode(). While this could be done, we're choosing not to
-    // support this now since it would require different logic around de-duping.
-    if (!(this.renderRoot instanceof window.ShadowRoot)) {
-      return;
-    }
-    if (supportsAdoptingStyleSheets) {
-      (this.renderRoot as ShadowRoot).adoptedStyleSheets = styles.map((s) =>
-        s instanceof CSSStyleSheet ? s : s.styleSheet!
-      );
-    } else {
-      styles.forEach((s) => {
-        const style = document.createElement('style');
-        style.textContent = (s as CSSResult).cssText;
-        this.renderRoot.appendChild(style);
-        (this._renderOptions as {
-          renderBefore: ChildNode;
-        }).renderBefore ??= style;
-      });
+  protected createRenderRoot() {
+    const renderRoot = super.createRenderRoot();
+    if (!supportsAdoptingStyleSheets) {
+      (this._renderOptions as {
+        renderBefore: ChildNode;
+      }).renderBefore ??= renderRoot!.lastChild as ChildNode;
     }
+    return renderRoot;
```

Comment:
Could this be simplified to:

```suggestion
  protected createRenderRoot() {
    const renderRoot = super.createRenderRoot();
    (this._renderOptions as {
        renderBefore: ChildNode;
      }).renderBefore ??= renderRoot!.firstChild;
    }
    return renderRoot;
```

If we always render to the first thing in the container, we'll go before emulated adoptedStyleSheets.

Also, is there no way around that cast? Maybe `this._renderOptions.renderBefore! ?? =`

Reply:
Fixed. Changed from a readonly property to allow stuff like this.